src/types/provider.ts

```diff
@@ -1,24 +1,19 @@
+/**
+ * Common interface response
+ * Intersection (sequencer response ∩ (∪ rpc responses))
+ */
 import BN from 'bn.js';
 
 import { Abi, CompressedProgram, EntryPointsByType, RawCalldata, Signature, Status } from './lib';
 
 export interface GetBlockResponse {
-  accepted_time: number;
+  timestamp: number;
   block_hash: string;
   block_number: number;
-  gas_price: string;
   new_root: string;
-  old_root?: string;
   parent_hash: string;
-  sequencer: string;
   status: Status;
   transactions: Array<string>;
-  starknet_version?: string;
-}
-
-export interface GetCodeResponse {
-  bytecode: string[];
```

Comment:
we heavily rely on this to detect if a contract is deployed.

I would suggest to keep it in the common interface and throw "Not implemented" error in rpc provider

Reply:
Hm,
Can we create **isContractDeployed** in a common interface that would return true/false, and then RPC can implement it with function 'starknet_getClassHashAt' that uses the same parameters contact address & blockIdentifier, and sequencer can implement it with getCode ?
https://github.com/starkware-libs/starknet-specs/blob/30e5bafcda60c31b5fb4021b4f5ddcfc18d2ff7d/api/starknet_api_openrpc.json#L267

Reply:
I would suggest to simply keep the method so dapps/wallets dont need to use a new api

RPC can throw as I said

In the mid-term this method should be dropped and both sequencer and rpc should just work with the `starknet_getClassHashAt` pattern